src/app/ui/file_selector.cpp

```diff
@@ -411,13 +411,9 @@ bool FileSelector::show(
   fs->refresh();
 
   // We have to find where the user should begin to browse files
-  std::string start_folder_path;
-  if (initialPath.empty()) {
-    start_folder_path = get_initial_path_to_select_filename(initialPath);
-  }
-  else {
-    start_folder_path = base::get_file_path(initialPath);
-  }
+  std::string start_folder_path = base::get_file_path(initialPath);
+  if (start_folder_path.empty())
+    start_folder_path = get_current_dir_for_file_selector();
```

Comment:
We can use `get_initial_path_to_select_filename()` to get the initial path+filename path, use the following code which has the same effect as the proposed fix:
```suggestion
  std::string start_folder_path =
    base::get_file_path(
      get_initial_path_to_select_filename(initialPath));
```